test/integration/admin_dashboard_test.rb

```diff
@@ -0,0 +1,27 @@
+require "test_helper"
+
+class AdminDashboardTest < ActionDispatch::IntegrationTest
+
+  test 'logged in admin sees dashboard page' do
+    admin = User.create(username: "admin",
+                        password: "password",
+                        role: 1)
+
+    ApplicationController.any_instance.stubs(:current_user).returns(admin)
+    visit admin_dashboard_path(admin)
+
+    assert page.has_content?("Welcome, Admin!")
+  end
+  test 'default user does not see admin dashboard' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.